javatests/arcs/core/storage/driver/DatabaseDriverTest.kt

```diff
@@ -348,18 +364,29 @@ class DatabaseDriverTest {
     assertThat(receiverVersion).isEqualTo(1)
   }
 
+  @Test
+  fun send_throws_ifUnsupportedDataType() = runBlockingTest {
+    val driver = buildDriver<Int>(database)
+
+    assertFailsWith(
+      UnsupportedOperationException::class,
+      "Unsupported type for DatabaseDriver: ${Int::class}"
+    ) {
+      driver.send(12, 1)
+    }
+  }
+
   @Test
   fun deletedAtDatabase_heardByDriver() = runBlockingTest {
     val driver = buildDriver<CrdtEntity.Data>(database)
     val entity = createPersonCrdt("jason", setOf("555-5555"))
+    driver.registerReceiver { data, _ -> assertThat(data).isNotNull() }
 
     driver.send(entity, 1)
 
-    assertThat(driver.getDatabaseData().first).isNotNull()
-
     database.delete(driver.storageKey)
 
-    assertThat(driver.getDatabaseData().first).isNull()
+    driver.registerReceiver { data, _ -> assertThat(data).isNull() }
```

Comment:
this does not make sure the driver is called, it only guarantees that if it is called, then data is null. Instead you can do like the other tests
var receiverData: CrdtSet.Data<Reference>? = something_not_null
driver.registerReceiver { data, _ ->
      receiverData = data
}
assertThat(receiverData).isNull()